### DISCUSSION TITLE
Security

### DISCUSSION THREAD

**Discussion by tieryo** [megacosm-entertainment/sentiencemud#5] [Ideas]
### Current Status:
We telnet. What security?
### Solution:
TLS support. We can use letsencrypt or a static certificate to secure traffic to and from the game. Not sure if we should have a second optional port, or support something like the STARTTLS command to switch to encrypted communication if clients request it on the same port.

---
### Current Status:
Passwords. sha256 encrypted, but just passwords, and still telnet.
### Solution:
Add in 2fa, password recovery that doesn't require admin intervention.

**Comment by ryanrhanson** [comment]
TLS is now supported on the 2.0 branches, with config options to start on both secure and plaintext ports.

Additionally, there is now a prompt option `%-` to show if you are currently on a secure or insecure port.

Finally, password recovery via email as also functional.

Next up is rounding out the 2fa additions to provide recovery codes, and email the qr code to users in addition to displaying it in the terminal.
